File: TaskManager.DAL/Models/Employee.cs
﻿using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Text;

namespace TaskManager.DAL.Models
{
    public class Employee
    {
        public Guid EmployeeId { get; set; }
        [Required]
        [MaxLength(20, ErrorMessage = "Login must be less than 20 chars long")]
        public string Login { get; set; }
        [Required]
        public string Password { get; set; }
        public virtual ICollection<EmployeeTeam> TeamsWithThisEmployee { get; set; }
        public virtual ICollection<EmployeeTask> TasksFromEmployee { get; set; }


    }
}

File: TaskManager-Client/src/app/model/employee.ts
import { Task } from './task';

export interface Employee {
    employeeId:string,
    login:string,
}
export interface Team{
    teamId:string,
    name:string,
    teamMembers:EmployeeTeam[]
}
export interface EmployeeTeam{
    employeeId:string,
    employee: Employee,
    teamId:string,
    team:Team
}
export interface EmployeeTask{
    employeeId:string,
    employee: Employee,
    taskId:string,
    task:Task
}